packages/cli/src/modules/insights/insights.service.ts

```diff
@@ -33,25 +34,45 @@ export class InsightsService {
 		private readonly collectionService: InsightsCollectionService,
 		private readonly pruningService: InsightsPruningService,
 		private readonly licenseState: LicenseState,
+		private readonly instanceSettings: InstanceSettings,
 		private readonly logger: Logger,
 	) {
 		this.logger = this.logger.scoped('insights');
 	}
 
 	startTimers() {
-		this.compactionService.startCompactionTimer();
+		if (
+			this.instanceSettings.instanceType !== 'main' &&
+			this.instanceSettings.instanceType !== 'webhook'
+		) {
+			return;
+		}
+
 		this.collectionService.startFlushingTimer();
+		this.logger.debug('Started collection flushing scheduler');
+
+		// Start compaction and pruning timers for main leader instance only
+		if (this.instanceSettings.isLeader) {
+			this.startCompactionAndPruningSchedulers();
+		}
+	}
+
+	@OnLeaderTakeover()
+	startCompactionAndPruningSchedulers() {
+		this.compactionService.startCompactionTimer();
+		this.logger.debug('Started compaction scheduler');
 		if (this.pruningService.isPruningEnabled) {
 			this.pruningService.startPruningTimer();
+			this.logger.debug('Started pruning scheduler');
 		}
-		this.logger.debug('Started compaction, flushing and pruning schedulers');
 	}
 
+	@OnLeaderStepdown()
 	stopTimers() {
 		this.compactionService.stopCompactionTimer();
-		this.collectionService.stopFlushingTimer();
+		this.logger.debug('Stopped compaction scheduler');
```

Comment:
The naming of the method is a rather misleading as now here we're never stopping the flushing timer.